Makefile

```diff
@@ -124,31 +126,40 @@ launch-dcos: check-terraform
 	cd .deploy; \
 	$(TERRAFORM_CMD) apply $(TERRAFORM_APPLY_ARGS) -var-file desired_cluster_profile
 
-.PHONY: kubectl-tunnel
-kubectl-tunnel:
-	$(KUBECTL_CMD) config set-cluster dcos-k8s --server=http://localhost:9000
-	$(KUBECTL_CMD) config set-context dcos-k8s --cluster=dcos-k8s --namespace=default
-	$(KUBECTL_CMD) config use-context dcos-k8s
-	$(call get_master_ip)
-	ssh -4 -o "UserKnownHostsFile=/dev/null" -o "StrictHostKeyChecking=no" -o "ServerAliveInterval=120" \
-		-N -L 9000:apiserver-insecure.kubernetes.l4lb.thisdcos.directory:9000 \
-		$(SSH_USER)@$(MASTER_IP)
+.PHONY: plan
+plan: plan-dcos
+
+.PHONY: deploy
+deploy: check-cli launch-dcos setup-cli install
+
+.PHONY: setup-cli
+setup-cli: check-dcos
+	@echo "Sleep for 15 seconds ..."
+	@sleep 15
+	$(call get_master_lb_ip)
+	$(DCOS_CMD) cluster setup https://$(MASTER_LB_IP)
 
 .PHONY: ui
 ui:
 	$(call get_master_lb_ip)
 	$(OPEN) https://$(MASTER_LB_IP)
 
-.PHONY: kube-ui
-kube-ui:
-	$(call get_master_lb_ip)
-	$(OPEN) https://$(MASTER_LB_IP)/service/kubernetes-proxy/
+.PHONY: install
+install: check-dcos
+	$(DCOS_CMD) package install --yes marathon-lb;\
+	$(DCOS_CMD) package install --yes kubernetes --package-version=$(KUBERNETES_FRAMEWORK_VERSION) --options=./.deploy/options.json;\
+	$(DCOS_CMD) marathon app add ./.deploy/kubeapi-proxy.json
 
-.PHONY: plan
-plan: plan-dcos
+.PHONY: watch
+watch:
+	watch dcos kubernetes plan show deploy
 
-.PHONY: deploy
-deploy: check-cli launch-dcos setup-cli install
+.PHONY: kubeconfig
+kubeconfig:
+	$(call get_public_agent_ip)
+	$(DCOS_CMD) kubernetes kubeconfig --apiserver-url https://$(PUBLIC_AGENT_IP):6443 --insecure-skip-tls-verify
+	@echo "Sleep for 25 seconds ..."
+	@sleep 25
```

Comment:
@rimusz why do we need to sleep?

Reply:
same here, better put some delay, as otherwise kubeapi is not available right after the above command

Reply:
Can we poll until the Kubernetes API is up instead?